Add new experimental picture viewer

nielsvanvelzen opened the pull request:
I started work on this around 2 months ago but didn't completely finish it yet. It already uses a feature flag which is disabled in releases so it won't hurt to merge it. I've made a new column in my board (https://github.com/jellyfin/jellyfin-androidtv/projects/15#column-19021853) that shows the remaining tasks.

The biggest feature is that it doesn't depend on the media manager anymore so it's sort of a part of the playback rewrite (#1057).

**Changes**
- Add preference to enable new picture viewer
  - Like the playback rewrite; only shows in development builds
- Add KeyHandler DSL to write readable key press handlers
- Add new picture viewer
  - Supports portrait mode
  - Sort of supports touch
  - Supports slideshows
  - Shows controls to make it easy to start slideshows and navigate forward/backward

**Issues**
<!-- Tag any issues that this PR solves here.
ex. Fixes # -->

thornbill commented on app/src/main/java/org/jellyfin/androidtv/ui/picture/PictureViewerFragment.kt:
```diff
@@ -0,0 +1,156 @@
+		/**
+		 * The download API is used by jellyfin-web when loading images. Unfortunately with our
+		 * current code it doesn't work for the app. Larger files (gifs, large photos etc.) can
+		 * cause the app to go out of memory. This is mostly catched by Glide but it ends up
```
catched -> caught?

thornbill commented on app/src/main/java/org/jellyfin/androidtv/ui/picture/PictureViewerViewModel.kt:
```diff
@@ -0,0 +1,103 @@
+		_currentItem.value = itemResponse
+
+		val albumResponse by api.itemsApi.getItemsByUserId(
+			parentId = itemResponse.albumId,
```
Album refers to music albums, so I think you would want to do something like the following:

```suggestion
			parentId = if (itemResponse.type === BaseItemKind.PHOTO_ALBUM) itemResponse.id else itemResponse.parentId,
```

nielsvanvelzen replied:
In the createIntent() function we check if the given BaseItem is PHOTO so it can never by PHOTO_ALBUM in this place.

thornbill reviewed (commented):
I'm not sure how perfect you are striving to get this for the first implementation, but I noticed a few issues:

1. There seems to be an issue when changing photos (either manually or via the slideshow) with gifs. The image will change briefly then go back to displaying the previous image.
2. In large photo libraries it seems like the current "position" is getting lost somehow. For example, I started playback on photo 301, click next, and a previous photo is displayed instead of photo 302.

It would probably be a good idea to show the photo name at least when the controls are visible (this would also help diagnose issue 2 further since I have many copies of the same photo in that album).

It might be good to show the "playback" controls by default to help with discoverability and hide them after a short time. I also didn't find pushing "up" to show the controls intuitive... I'm more accustomed to services using the "down" button to display them. That is more of a personal preference probably and there is some inconsistency among other services on that.

nielsvanvelzen commented:
> There seems to be an issue when changing photos (either manually or via the slideshow) with gifs. The image will change briefly then go back to displaying the previous image.

I haven't experienced this issue but I'd say it's out of scope for this initial PR.

> In large photo libraries it seems like the current "position" is getting lost somehow. For example, I started playback on photo 301, click next, and a previous photo is displayed instead of photo 302.

I'll need to create a bigger album (biggest is about 40 items right now) to test this but I've already made an task for lazy loading which might fix this issue.

> It would probably be a good idea to show the photo name at least when the controls are visible (this would also help diagnose issue 2 further since I have many copies of the same photo in that album).

Agreed, this was one of the things I wanted to add but not in this initial PR. This one too is already a task on the big list, it's not a feature in the current viewer either (except if you count that browse thingy that you can open while viewing).

> It might be good to show the "playback" controls by default to help with discoverability and hide them after a short time. I also didn't find pushing "up" to show the controls intuitive... I'm more accustomed to services using the "down" button to display them. That is more of a personal preference probably and there is some inconsistency among other services on that.

Added support for KEYCODE_DPAD_DOWN to open the controls, I think opening the controls makes sense when the timer to auto-close it is added. I've updated the task to describe this behavior.
